### PR TITLE
Adds Resampler to API docs

### LINKED ISSUE #5548: Add API docs for Resampler class
This may be a discoverability issue but I could not find API docs for the Resample class at https://docs.dask.org/en/latest/dataframe-api.html or https://docs.dask.org/en/latest/api.html

Screenshot from 2019-10-31 18-03-15

---

Separately, but relatedly, I find the information architecture around the API docs a little confusing. Typically, when I click on the API part of docs, I expect to go to an arguably less user-friendly part of a docs website and just be able to click around all classes and methods that are available. In the dask docs, when I click on the API link in the left menu, I get taken to a curated page. If I then click on a sub area (e.g. DataFrame) I am put back into the context of user documentation.

Some examples of docs where I have developed the expectation for this pattern:
* bokeh user guide vs reference (https://docs.bokeh.org/en/latest/docs/reference.html)
* jinja2 - https://jinja.palletsprojects.com/en/2.10.x/api/
* pandas - api reference https://pandas.pydata.org/pandas-docs/stable/reference/index.html
* spark - https://spark.apache.org/docs/latest/api/python/index.html (vs programming guides https://spark.apache.org/docs/latest/sql-getting-started.html)

Conversely, djangoproject is an example of one of, in my opinion, the best documented projects out there that doesn't meet this pattern. Also sphinx. This is just to emphasize that I'm not trying to claim that there are no standards and this separate observation is just that.

### PR DESCRIPTION
**Pull request by jrbourbeau**
Closes #5548

### CODE REVIEW

File: docs/source/dataframe-api.rst [hunk 1]
```diff
@@ -414,6 +414,15 @@ Rolling
 
 .. autofunction:: map_overlap
 
+Resampling
+~~~~~~~~~~
+
+.. currentmodule:: dask.dataframe.tseries.resample
+
+.. autoclass:: Resampler
```
Comment (mrocklin) [inline docs/source/dataframe-api.rst]: This class doesn't have a docstring, neither does the `_agg` method which seems to be used to generalize things.  We might want to add something small there, and also maybe make the `._agg` method non-private.

Comment (jrbourbeau) [reply docs/source/dataframe-api.rst]: Just pushed a commit which adds docstrings for `Resampler` and `Resampler._agg`. It seems the public `Resampler.agg` method should work for specifying general aggregations

```python
In [1]: from dask.dataframe.io.demo import make_timeseries

In [2]: ddf = make_timeseries()

In [3]: import numpy as np

In [4]: ddf.resample('1d').agg(np.sum).compute()
Out[4]:
                   id          x           y
timestamp
2000-01-31  8638696.0  47.077070 -134.875168
2000-02-01  8636932.0  78.899947  -22.346659
2000-02-02  8636624.0  32.193012  -18.677558
2000-02-03  8638606.0  70.468358  -10.982665
2000-02-04  8639956.0  44.161050  -77.516730
...               ...        ...         ...
2000-12-27  8639431.0  12.466025   62.642564
2000-12-28  8638456.0 -34.224802   57.942250
2000-12-29  8643282.0  20.416100   20.420697
2000-12-30  8643198.0  33.595944  -10.400613
2000-12-31        NaN        NaN         NaN

[336 rows x 3 columns]
```

Comment (mrocklin) [reply docs/source/dataframe-api.rst]: Oh grand

**Comment by mrocklin** [comment]
Thanks @jrbourbeau !

**Pull request merged**